files/private-chef-ctl-commands/key-control.rb

```diff
@@ -14,345 +14,428 @@
 # limitations under the License.
 #
 
-require "sequel"
 require "openssl"
-require "time"
+require "optparse"
+require "ostruct"
+require "chef/config"
+require "chef/rest"
 
 cmd_args = ARGV[3..-1]
 
-##################
-# Helper Classes #
-##################
-class Database
-  attr_reader :db
-  def initialize(running_config)
-    if running_config.nil?
-      STDERR.puts "Error: you must run chef-server-ctl reconfigure before using key commands."
-      exit 1
-    end
-
-    # Pull all the DB params from config; the database name seems to be hardcoded,
-    # however -- I found it in the postgresql.rb recipe in the private-chef cookbooks;
-    # I'm not sure why we hardcode it instead of making it configurable, but I assume
-    # there are reasons?
-    db_user = running_config['private_chef']['postgresql']['sql_user']
-    db_password = running_config['private_chef']['postgresql']['sql_password']
-    db_host = running_config['private_chef']['postgresql']['listen_address']
-
-    # Set convert_infinite_timestamps to a float so that Sequel
-    # doesn't crash out on querying an infinity timestamp.
-    # See this commit for more detail:
-    # https://github.com/jeremyevans/sequel/commit/2440e98b4a7063a90d8337e57b1c3e3d3a1cbd0b
-    @db = Sequel.connect(:adapter => 'postgres', :host => db_host,
-                         :database => 'opscode_chef', :user => db_user,
-                         :password => db_password, :convert_infinite_timestamps => :float)
-
-  end
+pivotal_config = "/etc/opscode/pivotal.rb"
+if File.exist?(pivotal_config)
+  Chef::Config.from_file(pivotal_config)
+  @chef_rest = Chef::REST.new(Chef::Config[:chef_server_root])
+  @valid_iso_date = /^(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z|infinity)$/
 end
 
-class PublicKeyOutputter
-  def output_key_results(results, hide_public_keys, database)
-    results.each do |row|
-      # get expiration date from keys table as
-      # results variable is result of keys_by_name view
-      expires_at = database.db[:keys].where(:id => row[:id]).where(:key_name => row[:key_name]).first[:expires_at]
-      puts "\nkey_name: #{row[:key_name]}"
-      puts "expires_at: #{expires_at}"
-      unless hide_public_keys
-        puts "public_key:"
-        puts "#{row[:public_key]}"
+module KeyCtlHelper
+  # Optparse doesn't properly handle the case where you specify an argument with mandatory input
```

Comment:
Is this actually true still, even after the changes? I'd suggest checking in with other teams that have used `optparse` more extensively.